posthog/api/test/test_decide.py

```diff
@@ -240,15 +240,15 @@ def test_feature_flags(self):
             created_by=self.user,
         )
 
-        with self.assertNumQueries(1):
+        with self.assertNumQueries(4):
```

Comment:
this is _still_ one query, with 3 surrounding queries that should be very cheap. (artifact of creating a multi-line transaction). Will monitor decide performance once this goes in.

The 3 extra queries (you can find these in the snapshot tests as well):

SAVEPOINT.
SET LOCAL statement_timeout
..<actual query!>
RELEASE SAVEPOINT

Reply:
And I'm guessing this is using SAVEPOINTs because the test is already wrapping the test in a BEGIN...ROLLBACK?

Reply:
Correct.

Actually, I think I can remove this if I use a `TransactionTestCase` instead. Getting an accurate picture here is probably more important than speed. Let me see 👀

Reply:
yeah, that just breaks everything :P (`client.force_login()` fails with django.db.utils.DatabaseError: Save with update_fields did not affect any rows. )